pkg/controller/provider/openstack/designateclient.go

```diff
@@ -66,25 +68,40 @@ type authConfig struct {
 	AuthURL     string
 	Username    string
 	DomainName  string
+	DomainID    string
 	Password    string
 	ProjectName string
+	ProjectID   string
+	// UserDomainName/ID are optional
+	UserDomainID   string
+	UserDomainName string
 	// RegionName is optional
 	RegionName string
 }
 
 // authenticate in OpenStack and obtain Designate service endpoint
 func createDesignateServiceClient(logger logger.LogContext, authConfig *authConfig) (*gophercloud.ServiceClient, error) {
-	opts := gophercloud.AuthOptions{
-		IdentityEndpoint: authConfig.AuthURL,
-		Username:         authConfig.Username,
-		Password:         authConfig.Password,
-		TenantName:       authConfig.ProjectName,
-		DomainName:       authConfig.DomainName,
-		AllowReauth:      true,
+	clientOpts := new(clientconfig.ClientOpts)
+	authInfo := &clientconfig.AuthInfo{
+		AuthURL:        authConfig.AuthURL,
+		Username:       authConfig.Username,
+		Password:       authConfig.Password,
+		DomainName:     authConfig.DomainName,
+		DomainID:       authConfig.DomainID,
+		ProjectName:    authConfig.ProjectName,
+		ProjectID:      authConfig.ProjectID,
+		UserDomainName: authConfig.UserDomainName,
+		UserDomainID:   authConfig.UserDomainID,
 	}
+	clientOpts.AuthInfo = authInfo
 
-	logger.Infof("Using OpenStack Keystone at %s", opts.IdentityEndpoint)
-	authProvider, err := openstack.NewClient(opts.IdentityEndpoint)
+	ao, err := clientconfig.AuthOptions(clientOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client auth options: %+v", err)
+	}
+
```

Comment:
Please add the `AllowReauth` option again here:
```go
	ao.AllowReauth = true
```

Reply:
Good point!